src/components/ConditionalFilter/Validation/metadata.ts

```diff
@@ -0,0 +1,25 @@
+import { FilterElement } from "../FilterElement";
+
+export const metadata = (element: FilterElement, row: number) => {
+  const { value } = element.condition.selected;
+
+  // Metadata value should be a tuple [key, value]
+  if (!Array.isArray(value) || value.length !== 2) {
+    return {
+      row,
+      rightText: "Invalid metadata format",
+    };
+  }
+
+  const [key] = value as [string, string];
+
+  // Key is always required
+  if (!key || key.trim() === "") {
+    return {
+      row,
+      rightText: "Metadata key is required",
+    };
+  }
+
+  return false;
+};
```

Comment:
please avoid such cryptic functions. `metadata()` doesnt mean anything. Function has no typed return statement so we need to scan implementation what it does. Finally I see `metadata() === false` which is even more confusing

also if this is validator, common pattern is returning errors or null / undefined (and lets type function return statement)